pkg/test/executor/docker_entrypoint_test.go

```diff
@@ -0,0 +1,281 @@
+//go:build integration || !unit
+
+package executor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/bacalhau-project/bacalhau/pkg/docker"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/storage/util"
+	"github.com/bacalhau-project/bacalhau/pkg/test/scenario"
+	"github.com/bacalhau-project/bacalhau/pkg/util/targzip"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+//When the entrypoint flag is used on the CLI, eg "docker run --entrypoint /bin/echo image hello world" docker will ignore the CMD
+//Stored in the dockerfile. Any paramaters used after the image will be interpreted as the new CMD.
+//If no entrypoint is specified in the dockerfile and the CMD appended to the CLI does not contain an executable that can
+//be found in Alpine's $PATH, the docker daemon will throw an error.
+//Please note that if a dockerfile specifies neither the CMD or ENTRYPOINT then docker will use the base image's (if specified).
+//Alpine's base image has a CMD of '/bin/sh' and no entrypoint.
```

Comment:
This should be true of all images, not just alpine. So perhaps replace references to "Alpine" with "the chosen image"?